src/ArmTemplates/Extractor/EntityExtractors/GatewayApiExtractor.cs

```diff
@@ -108,14 +116,11 @@ List<GatewayApiTemplateResource> GenerateGatewayApiTemplateResources(string gate
                 gatewayApiTemplateResource.Name = $"[concat(parameters('{ParameterNames.ApimServiceName}'), '/{gatewayName}/{api.Name}')]";
                 gatewayApiTemplateResource.ApiVersion = GlobalConstants.ApiVersion;
                 gatewayApiTemplateResource.Scale = null;
-                gatewayApiTemplateResource.DependsOn = string.IsNullOrEmpty(dependsOn) ? Array.Empty<string>() : new[] { dependsOn };
                 gatewayApiTemplateResource.Properties = new GatewayApiProperties 
                 {
                     ProvisioningState = "created" 
                 };
 
-                dependsOn = $"[resourceId('Microsoft.ApiManagement/service/gateways/apis', parameters('{ParameterNames.ApimServiceName}'), '{gatewayName}', '{api.Name}')]";
```

Comment:
Shouldn't we keep this; but movei t up?

Reply:
There is no need to put dependsOn for the gatewayApis resource, since it could be uploaded in parallel while template deployment